Doc/library/contextlib.rst

```diff
@@ -137,6 +137,21 @@ Functions and classes provided:
    ``page.close()`` will be called when the :keyword:`with` block is exited.
 
 
+.. function:: nullcontext(thing=None)
+
+    Return a context manager that just returns *thing*. It is intended to be used
+    as a standin for an optional context manager, for example::
+
+        def debug_trace(details):
+            if __debug__:
+                return TraceContext(details)
+            # Don't do anything special with the context in release mode
+            return nullcontext(details)
+
+        with debug_trace(details) as d:
+            # Suite is traced in debug mode, but runs normally otherwise
```

Comment:
Can you check if this code block is rendered correctly?  Since comments are stripped out and do not become statements, this may be invalid code.  If that is so (I’m not sure!), adding an ellipsis would fix it.

Reply:
It is rendered correctly. I am unsure about the example though. I used the example previously used for the recipe using ExitStack and could not think of a better one. I don't find it all too intuitive though.

Reply:
Maybe there was a better use case in the ticket?

Reply:
An example I encountered was having to close a file if I opened it in the function, but not if a file object was passed to the function.  This could read:

```python

def function(target):
    """"Do something with target (file object or path as string."""
    if isinstance(target, str):   # or FileLike
        cm = open(target)
    else:
        # Wrap the file in a no-op context manager, the caller will close it
        cm = nullcontext(target)

    with cm as file:
        print('hello', file=file)
```

Reply:
Thanks for the suggestion. Changed the docs.